lib/auth/auth.go

```diff
@@ -3864,6 +3864,49 @@ func (a *Server) CreateAccessRequest(ctx context.Context, req types.AccessReques
 		return trace.Wrap(err)
 	}
 
+	// Look for user groups and associated applications to the request.
+	requestedResourceIDs := req.GetRequestedResourceIDs()
+	var additionalResources []types.ResourceID
+
+	var userGroups []types.ResourceID
+	existingApps := map[string]struct{}{}
+	for _, resource := range requestedResourceIDs {
+		switch resource.Kind {
+		case types.KindApp:
+			existingApps[resource.Name] = struct{}{}
+		case types.KindUserGroup:
+			userGroups = append(userGroups, resource)
+		}
```

Comment:
Why do we only handle Applications and Groups? Why don't we handle other resource types here?

Reply:
At the moment these are the only resources that have this sort of relationship. I'm guessing that if we decide to do something similar elsewhere we should expand this.